sites/partners/pages/sign-in.tsx

```diff
@@ -1,39 +1,148 @@
-  const { login } = useContext(AuthContext)
+  const { login, requestMfaCode } = useContext(AuthContext)
   /* Form Handler */
   // eslint-disable-next-line @typescript-eslint/unbound-method
-  const { register, handleSubmit, errors } = useForm()
+  const { register, handleSubmit, errors, setValue, control } = useForm()
   const { networkError, determineNetworkError, resetNetworkError } = useCatchNetworkError()
   const router = useRouter()
+  const [email, setEmail] = useState<string | undefined>(undefined)
+  const [password, setPassword] = useState<string | undefined>(undefined)
+  const [mfaType, setMfaType] = useState<EnumRequestMfaCodeMfaType | undefined>(undefined)
+  const [renderStep, setRenderStep] = useState<EnumRenderStep | undefined>(
+    EnumRenderStep.emailAndPassword
+  )
 
-  const onSubmit = async (data: { email: string; password: string }) => {
-    const { email, password } = data
+  const onSubmitEmailAndPassword = useCallback(
+    async (data: { email: string; password: string }) => {
+      const { email, password } = data
+      try {
+        await login(email, password)
+        await router.push("/")
+      } catch (error) {
+        if (error?.response?.data?.name === "mfaCodeIsMissing") {
+          setEmail(email)
+          setPassword(password)
+          setRenderStep(EnumRenderStep.mfaType)
+        } else {
+          const { status } = error.response || {}
+          determineNetworkError(status, error)
+        }
+      }
+    },
+    [setEmail, setPassword, setRenderStep, determineNetworkError, login, router]
+  )
 
-    try {
-      await login(email, password)
-      await router.push("/")
-    } catch (error) {
-      const { status } = error.response || {}
-      determineNetworkError(status, error)
-    }
-  }
-
-  return (
-    <FormsLayout>
-      <FormSignIn
-        onSubmit={onSubmit}
-        control={{ register, errors, handleSubmit }}
-        networkError={{ error: networkError, reset: resetNetworkError }}
-      />
-    </FormsLayout>
+  const onSubmitMfaType = useCallback(
+    async (data: { mfaType: EnumRequestMfaCodeMfaType }) => {
+      const { mfaType: incomingMfaType } = data
+      try {
+        await requestMfaCode(email, password, incomingMfaType)
+        setMfaType(incomingMfaType)
+        setRenderStep(EnumRenderStep.enterCode)
+      } catch (error) {
+        if (error?.response?.data?.name === "phoneNumberMissing") {
+          setMfaType(incomingMfaType)
+          setRenderStep(EnumRenderStep.phoneNumber)
+        }
+      }
+    },
+    [email, password, setMfaType, setRenderStep, requestMfaCode]
+  )
+
+  const onSubmitMfaCodeWithPhone = useCallback(
+    async (data: { phoneNumber: string }) => {
+      const { phoneNumber } = data
+      await requestMfaCode(email, password, mfaType, phoneNumber)
+      setRenderStep(EnumRenderStep.enterCode)
+    },
+    [email, password, mfaType, setRenderStep, requestMfaCode]
+  )
+
+  const onSubmitMfaCode = useCallback(
```

Comment:
It may make sense to move the onSubmitMfa handlers to a separate mfa helpers file under lib